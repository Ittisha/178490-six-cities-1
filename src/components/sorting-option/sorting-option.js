import * as React from 'react';
import PropTypes from 'prop-types';

import {SortingOptions} from '../../consts';

class SortingOption extends React.PureComponent {
  constructor(props) {
    super(props);

    this._handleOptionClick = this._handleOptionClick.bind(this);
  }

  render() {
    const {
      option,
      isActive,
    } = this.props;
    const activeClass = isActive ? `places__option--active` : ``;

    return <li
      className={`places__option ${activeClass}`}
      tabIndex={0}
      onClick={this._handleOptionClick}
    >
      {SortingOptions[option]}
    </li>;
  }

  _handleOptionClick() {
    const {
      option,
      onOptionSelect,
    } = this.props;

    onOptionSelect(option);
  }
}

SortingOption.propTypes = {
  option: PropTypes.string.isRequired,
  isActive: PropTypes.bool,
  onOptionSelect: PropTypes.func.isRequired,
};

export {SortingOption};
