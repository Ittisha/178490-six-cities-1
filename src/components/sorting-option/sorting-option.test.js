import * as React from 'react';
import renderer from 'react-test-renderer';

import {SortingOption} from './sorting-option';

const mockOption = {
  option: `Popular`,
  isActive: true,
};

it(`PlaceSortingType correctly renders`, () => {

  const onOptionSelect = jest.fn();
  const tree = renderer
    .create(<SortingOption
      option={mockOption.option}
      isActive={mockOption.isActive}
      onOptionSelect={onOptionSelect}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
