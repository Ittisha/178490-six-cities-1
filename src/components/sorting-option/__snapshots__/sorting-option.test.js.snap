// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceSortingType correctly renders 1`] = `
<li
  className="places__option places__option--active"
  onClick={[Function]}
  tabIndex={0}
/>
`;
